osquery/core/windows/wmi.h

```diff
@@ -173,8 +183,9 @@ class WmiRequest {
  private:
   Status status_;
   std::vector<WmiResultItem> results_;
-  IWbemLocator* locator_{nullptr};
-  IWbemServices* services_{nullptr};
-  IEnumWbemClassObject* enum_{nullptr};
+
+  unique_IWbemLocatorRef locator_{nullptr, free_IWbemLocatorRef};
+  unique_IEnumWbemClassObjectRef enum_{nullptr, free_IEnumWbemClassObjectRef};
+  shared_IWbemServicesRef services_{nullptr, free_IWbemServicesRef};
```

Comment:
May I ask you to implement it with lambdas instead?
Smth like:
```c++
namespace impl {

auto wbemObejectDeleter = [](auto* ptr) {
  ptr->Release();
  delete ptr;
};

}

class WmiRequest {
public:
    std::unique_ptr<IWbemLocator, decltype(impl::wbemObejectDeleter)>  locator_{nullptr, impl::wbemObejectDeleter};
    std::unique_ptr<IEnumWbemClassObject, decltype(impl::wbemObejectDeleter)> enum_{nullptr, impl::wbemObejectDeleter};
};
```

Reply:
sure - was the `delete ptr;` in your example lambda intentional?

Reply:
It was not intentional, sorry for the confusion, I don't know for sure how windows API works here, that was my guess.